lua/core/menu.lua

```diff
@@ -1145,7 +1146,104 @@ m.deinit[pRESET] = function() end
 
 
 -----------------------------------------
+-- UPDATE
+m.update = {}
+m.update.url = ''
+m.update.version = ''
+
+local function check_newest()
+  print("checking for update")
+  m.update.url = util.os_capture( [[curl -s \
+      https://api.github.com/repos/monome/norns/releases/latest \
+      | grep "browser_download_url.*" \
+      | cut -d : -f 2,3 \
+      | tr -d \"]])
+  print(m.update.url)
+  m.update.version = m.update.url:match("(%d%d%d%d%d%d)")
+  print("available version "..m.update.version)
+end
+
+local function get_update()
+  m.update.message = "preparing..."
+  menu.redraw()
+  pcall(cleanup) -- shut down script
+  norns.script.clear()
+  os.execute("sudo systemctl stop norns-jack.service") -- disable audio
+  os.execute("sudo rm -rf /home/we/update/*") -- clear old updates
+  m.update.message = "downloading..."
+  menu.redraw()
+  os.execute("wget -q -P /home/we/update/ " .. m.update.url) --download
+  m.update.message = "unpacking update..."
+  menu.redraw()
+  os.execute("tar xzvf /home/we/update/*.tgz -C /home/we/update/")
+  m.update.message = "running update..."
+  menu.redraw()
+  os.execute("/home/we/update/"..m.update.version.."/update.sh")
+  m.update.message = "complete."
+  menu.redraw()
+end
+
+m.key[pUPDATE] = function(n,z)
+  if m.update.stage==1 and z==1 then
```

Comment:
maybe consider constants for update stages to improve readability?  that is, if you did something like

```lua
local initializing, updating, getting_update, got_update = 1, 2, 3, 4
```

you could then do something like:

```lua
...

if m.update.stage == initializing and z == 1 then 
  ...
```

alternatively, you could emulate an enum i guess.  maybe like:

```lua
local stage = {
  initializing = 1,
  updating = 2,
}
```

and then 

```lua
if m.update.stage == stage.initializing and z == 1 then 
  ...
```

Reply:
hmmmm.... or maybe stage values are meaningful?  dunno. 🤔

Reply:
rad! i'll change the magic numbers to strings instead